openwebstart/src/main/java/com/openwebstart/mimetype/MimeTypeDetection.java

```diff
@@ -0,0 +1,46 @@
+package com.openwebstart.mimetype;
+
+import net.adoptopenjdk.icedteaweb.Assert;
+import net.adoptopenjdk.icedteaweb.logging.Logger;
+import net.adoptopenjdk.icedteaweb.logging.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PushbackInputStream;
+
+public class MimeTypeDetection {
```

Comment:
Maybe all the methods in this class can be moved to `MimeTypeInputStream`

Reply:
Since we have several unit tests for this class I would prefer to leave it in this class for now.

Reply:
... but methods are not public anymore